### ISSUE TITLE
module not found: error: can't resolve 'fs'

### ISSUE THREAD

**Issue by ChenWenjiang** [laurent22/joplin#4555]
I'm newcomer to JS/TS，this is my first JS/TS project.
I want to develop a plugin to export notebook by my way.
But when I use fs-extra module in my plugin，'npm run dist’ reports error like this:
```
chenwj@chenwj-Precision-3540:~/Projects/joplin$ npm run dist

> joplin-plugin-tohexo@1.0.0 dist /home/chenwj/Projects/joplin
> webpack --joplin-plugin-config buildMain && webpack --joplin-plugin-config buildExtraScripts && webpack --joplin-plugin-config createArchive

(node:11222) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
(Use `node --trace-deprecation ...` to show where the warning was created)
Plugin archive has been created in /home/chenwj/Projects/joplin/publish/com.peacedog.tohexo.jpl
Could not get git commit (not a git repo?): Command failed: git rev-parse --abbrev-ref HEAD
Git information will not be stored in plugin info file
Child

    ERROR in ./dist/index.js
    Module not found: Error: Can't resolve 'fs' in '/home/chenwj/Projects/joplin/dist'
     @ ./dist/index.js 1:9350-9363
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! joplin-plugin-tohexo@1.0.0 dist: `webpack --joplin-plugin-config buildMain && webpack --joplin-plugin-config buildExtraScripts && webpack --joplin-plugin-config createArchive`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the joplin-plugin-tohexo@1.0.0 dist script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/chenwj/.npm/_logs/2021-02-19T02_20_06_238Z-debug.log
```
I found a report to the same problem:https://github.com/laurent22/joplin/issues/4372, 
then I found there were 3 files in my project have configures like this 
```
  "browser": {
    "fs": false
  }
```
I changed them all, but not worked for me.

## Environment

Joplin version:  1.7.11(prod,linux)
Platform: Linux
OS specifics: Ubuntu18.04/20.04

## Steps to reproduce

1. Before I used fs-extra module, everything is OK.
2. I add `import * as fs from 'fs-extra';` and use fs.writeFileSync in my index.ts, it reports the problem. 

## Logfile

log is:
```
chenwj@chenwj-Precision-3540:~/Projects/joplin$ cat  /home/chenwj/.npm/_logs/2021-02-19T02_20_06_238Z-debug.log
0 info it worked if it ends with ok
1 verbose cli [
1 verbose cli   '/home/chenwj/.nvm/versions/node/v14.15.5/bin/node',
1 verbose cli   '/home/chenwj/.nvm/versions/node/v14.15.5/bin/npm',
1 verbose cli   'run',
1 verbose cli   'dist'
1 verbose cli ]
2 info using npm@6.14.11
3 info using node@v14.15.5
4 verbose run-script [ 'predist', 'dist', 'postdist' ]
5 info lifecycle joplin-plugin-tohexo@1.0.0~predist: joplin-plugin-tohexo@1.0.0
6 info lifecycle joplin-plugin-tohexo@1.0.0~dist: joplin-plugin-tohexo@1.0.0
7 verbose lifecycle joplin-plugin-tohexo@1.0.0~dist: unsafe-perm in lifecycle true
8 verbose lifecycle joplin-plugin-tohexo@1.0.0~dist: PATH: /home/chenwj/.nvm/versions/node/v14.15.5/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/chenwj/Projects/joplin/node_modules/.bin:/home/chenwj/.nvm/versions/node/v14.15.5/bin:/home/chenwj/.local/bin:/home/chenwj/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/chenwj/bin
9 verbose lifecycle joplin-plugin-tohexo@1.0.0~dist: CWD: /home/chenwj/Projects/joplin
10 silly lifecycle joplin-plugin-tohexo@1.0.0~dist: Args: [
10 silly lifecycle   '-c',
10 silly lifecycle   'webpack --joplin-plugin-config buildMain && webpack --joplin-plugin-config buildExtraScripts && webpack --joplin-plugin-config createArchive'
10 silly lifecycle ]
11 silly lifecycle joplin-plugin-tohexo@1.0.0~dist: Returned: code: 2  signal: null
12 info lifecycle joplin-plugin-tohexo@1.0.0~dist: Failed to exec dist script
13 verbose stack Error: joplin-plugin-tohexo@1.0.0 dist: `webpack --joplin-plugin-config buildMain && webpack --joplin-plugin-config buildExtraScripts && webpack --joplin-plugin-config createArchive`
13 verbose stack Exit status 2
13 verbose stack     at EventEmitter.<anonymous> (/home/chenwj/.nvm/versions/node/v14.15.5/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:315:20)
13 verbose stack     at ChildProcess.<anonymous> (/home/chenwj/.nvm/versions/node/v14.15.5/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:315:20)
13 verbose stack     at maybeClose (internal/child_process.js:1048:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
14 verbose pkgid joplin-plugin-tohexo@1.0.0
15 verbose cwd /home/chenwj/Projects/joplin
16 verbose Linux 5.4.0-65-generic
17 verbose argv "/home/chenwj/.nvm/versions/node/v14.15.5/bin/node" "/home/chenwj/.nvm/versions/node/v14.15.5/bin/npm" "run" "dist"
18 verbose node v14.15.5
19 verbose npm  v6.14.11
20 error code ELIFECYCLE
21 error errno 2
22 error joplin-plugin-tohexo@1.0.0 dist: `webpack --joplin-plugin-config buildMain && webpack --joplin-plugin-config buildExtraScripts && webpack --joplin-plugin-config createArchive`
22 error Exit status 2
23 error Failed at the joplin-plugin-tohexo@1.0.0 dist script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 2, true ]
```

**Comment by CalebJohn** [comment]
Hi @ChenWenjiang 

I'm glad to see your looking to develop a plugin for Joplin! 
Indeed native packages are not supported by plugins because plugins are required to work cross-platform. You can however use the [joplin.require](https://joplinapp.org/api/references/plugin_api/classes/joplin.html#require) function to import `fs-extra` (it also provides access to `sqlite3`).

I hope to see you release the plugin to the public one day!

**Comment by ChenWenjiang** [comment]
Thanks so much! @CalebJohn
I think your answer make me understand what is native package.
Also, I have another question when l solved this problem:
I used fs-extra package as https://github.com/laurent22/joplin/blob/dev/packages/app-cli/tests/support/plugins/nativeModule/src/index.ts , but it reports an error:
```
TS2339: Property 'require' does not exist on type 'JoplinPlugins'.
```
So, I copied https://github.com/laurent22/joplin/blob/dev/packages/app-cli/tests/support/plugins/nativeModule/src/index.ts to my project, but it also reports the error.
According the error, I found the file api/JoplinPlugins.d.ts has no `require`.
I copied the api folder replace in my project from https://github.com/laurent22/joplin/blob/dev/packages/app-cli/tests/support/plugins/nativeModule/api, now it's OK.

the content of files in api is different between the project  created by `yo joplin` and github repo.

**Comment by CalebJohn** [comment]
Is it possible that you initiated the project a few weeks back? The require function. Was added fairly recently.

**Comment by ChenWenjiang** [comment]
Just before a few minutes, I tried to make a new project by `yo joplin`,and then cp my plugin src to this project, same problem happend.
My operation is :
```
chenwj@chenwj-Precision-3540:~/Projects/test/joplintest$ yo joplin

     _-----_     
    |       |    ╭──────────────────────────╮
    |--(o)--|    │    Welcome to the fine   │
   `---------´   │ Joplin Plugin generator! │
    ( _´U`_ )    ╰──────────────────────────╯
    /___A___\   /
     |  ~  |     
   __'.___.'__   
 ´   `  |° ´ Y ` 

? Plugin ID
  Must be a globally unique ID such as "com.example.MyPlugin" or a UUID: com.test.test
? Plugin name
  User-friendly string which will be displayed in UI: test.test
? Plugin description: test test
? Author: a
? Repository URL: 
? Homepage URL: 
? The npm package will be named: "joplin-plugin-test-test"
  Press ENTER to keep this default, or type a name to change it: 
   create package.json
   create .gitignore
   create .npmignore
   create GENERATOR_DOC.md
   create tsconfig.json
   create webpack.config.js
   create plugin.config.json
   create src/index.ts
   create src/manifest.json
   create README.md
   create api/Global.d.ts
   create api/index.ts
   create api/Joplin.d.ts
   create api/JoplinCommands.d.ts
   create api/JoplinContentScripts.d.ts
   create api/JoplinData.d.ts
   create api/JoplinFilters.d.ts
   create api/JoplinInterop.d.ts
   create api/JoplinPlugins.d.ts
   create api/JoplinSettings.d.ts
   create api/JoplinViews.d.ts
   create api/JoplinViewsDialogs.d.ts
   create api/JoplinViewsMenuItems.d.ts
   create api/JoplinViewsMenus.d.ts
   create api/JoplinViewsPanels.d.ts
   create api/JoplinViewsToolbarButtons.d.ts
   create api/JoplinWorkspace.d.ts
   create api/types.ts


I'm all done. Running npm install for you to install the required dependencies. If this fails, try running the command yourself.


npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.

> joplin-plugin-test-test@1.0.0 prepare /home/chenwj/Projects/test/joplintest
> npm run dist


> joplin-plugin-test-test@1.0.0 dist /home/chenwj/Projects/test/joplintest
> webpack --joplin-plugin-config buildMain && webpack --joplin-plugin-config buildExtraScripts && webpack --joplin-plugin-config createArchive

(node:23167) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
(Use `node --trace-deprecation ...` to show where the warning was created)
Plugin archive has been created in /home/chenwj/Projects/test/joplintest/publish/com.test.test.jpl
Could not get git commit (not a git repo?): Command failed: git rev-parse --abbrev-ref HEAD
Git information will not be stored in plugin info file
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.3.1 (node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.2.7 (node_modules/watchpack-chokidar2/node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN joplin-plugin-test-test@1.0.0 No repository field.

added 515 packages from 279 contributors and audited 519 packages in 17.066s

22 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

chenwj@chenwj-Precision-3540:~/Projects/test/joplintest$ ls
api   GENERATOR_DOC.md  package.json       plugin.config.json  README.md  tsconfig.json
dist  node_modules      package-lock.json  publish             src        webpack.config.js
chenwj@chenwj-Precision-3540:~/Projects/test/joplintest$ cp ../joplin/src/
index.ts       manifest.json  
chenwj@chenwj-Precision-3540:~/Projects/test/joplintest$ cp ../joplin/src/index.ts src/index.ts 
chenwj@chenwj-Precision-3540:~/Projects/test/joplintest$ ls
api   GENERATOR_DOC.md  package.json       plugin.config.json  README.md  tsconfig.json
dist  node_modules      package-lock.json  publish             src        webpack.config.js
chenwj@chenwj-Precision-3540:~/Projects/test/joplintest$ npm run dist

> joplin-plugin-test-test@1.0.0 dist /home/chenwj/Projects/test/joplintest
> webpack --joplin-plugin-config buildMain && webpack --joplin-plugin-config buildExtraScripts && webpack --joplin-plugin-config createArchive

Child
    
    ERROR in /home/chenwj/Projects/test/joplintest/src/index.ts
    ./src/index.ts
    [tsl] ERROR in /home/chenwj/Projects/test/joplintest/src/index.ts(4,27)
          TS2339: Property 'require' does not exist on type 'JoplinPlugins'.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! joplin-plugin-test-test@1.0.0 dist: `webpack --joplin-plugin-config buildMain && webpack --joplin-plugin-config buildExtraScripts && webpack --joplin-plugin-config createArchive`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the joplin-plugin-test-test@1.0.0 dist script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/chenwj/.npm/_logs/2021-02-20T09_29_33_932Z-debug.log
chenwj@chenwj-Precision-3540:~/Projects/test/joplintest$ ls
api               node_modules  package-lock.json   publish    src            webpack.config.js
GENERATOR_DOC.md  package.json  plugin.config.json  README.md  tsconfig.json
chenwj@chenwj-Precision-3540:~/Projects/test/joplintest$ ll
total 252
drwxr-xr-x   6 chenwj chenwj   4096 Feb 20 17:29 ./
drwxr-xr-x  10 chenwj chenwj   4096 Feb 20 17:28 ../
drwxr-xr-x   2 chenwj chenwj   4096 Feb 20 17:28 api/
-rw-r--r--   1 chenwj chenwj   4029 Feb 20 17:28 GENERATOR_DOC.md
-rw-r--r--   1 chenwj chenwj     29 Feb 20 17:28 .gitignore
drwxr-xr-x 338 chenwj chenwj  12288 Feb 20 17:29 node_modules/
-rw-r--r--   1 chenwj chenwj     71 Feb 20 17:28 .npmignore
-rw-r--r--   1 chenwj chenwj    756 Feb 20 17:28 package.json
-rw-r--r--   1 chenwj chenwj 180595 Feb 20 17:29 package-lock.json
-rw-r--r--   1 chenwj chenwj     23 Feb 20 17:28 plugin.config.json
drwxr-xr-x   2 chenwj chenwj   4096 Feb 20 17:29 publish/
-rw-r--r--   1 chenwj chenwj   1271 Feb 20 17:28 README.md
drwxr-xr-x   2 chenwj chenwj   4096 Feb 20 17:28 src/
-rw-r--r--   1 chenwj chenwj    152 Feb 20 17:28 tsconfig.json
-rw-r--r--   1 chenwj chenwj   8759 Feb 20 17:28 webpack.config.js
```

**Comment by CalebJohn** [comment]
Please try `npm run update`
I think your generator is out of date.

**Comment by ChenWenjiang** [comment]
very strange:
```
chenwj@chenwj-Precision-3540:~/Projects/joplin$ npm run update

> joplin-plugin-tohexo@1.0.0 update /home/chenwj/Projects/joplin
> npm install -g generator-joplin && yo joplin --update

npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
+ generator-joplin@1.7.3
updated 5 packages in 42.522s

     _-----_     
    |       |    ╭──────────────────────────╮
    |--(o)--|    │    Welcome to the fine   │
   `---------´   │ Joplin Plugin generator! │
    ( _´U`_ )    ╰──────────────────────────╯
    /___A___\   /
     |  ~  |     
   __'.___.'__   
 ´   `  |° ´ Y ` 

? Updating will overwrite the config-related files. It will not change the
  content of /src or README.md. If you have made any changes to some of the
  config files make sure your code is under version control so that you can
  inspect the diff and re-apply your changes if needed. Do you want to proceed? Yes
    force package.json
    force .gitignore
    force .npmignore
    force GENERATOR_DOC.md
    force tsconfig.json
    force webpack.config.js
    force api/Global.d.ts
    force api/index.ts
    force api/Joplin.d.ts
    force api/JoplinCommands.d.ts
    force api/JoplinContentScripts.d.ts
    force api/JoplinData.d.ts
    force api/JoplinFilters.d.ts
    force api/JoplinInterop.d.ts
    force api/JoplinPlugins.d.ts
    force api/JoplinSettings.d.ts
    force api/JoplinViews.d.ts
    force api/JoplinViewsDialogs.d.ts
    force api/JoplinViewsMenuItems.d.ts
    force api/JoplinViewsMenus.d.ts
    force api/JoplinViewsPanels.d.ts
    force api/JoplinViewsToolbarButtons.d.ts
    force api/JoplinWorkspace.d.ts
    force api/types.ts


I'm all done. Running npm install for you to install the required dependencies. If this fails, try running the command yourself.



> joplin-plugin-tohexo@1.0.0 prepare /home/chenwj/Projects/joplin
> npm run dist


> joplin-plugin-tohexo@1.0.0 dist /home/chenwj/Projects/joplin
> webpack --joplin-plugin-config buildMain && webpack --joplin-plugin-config buildExtraScripts && webpack --joplin-plugin-config createArchive

Child
    
    ERROR in /home/chenwj/Projects/joplin/src/index.ts
    ./src/index.ts
    [tsl] ERROR in /home/chenwj/Projects/joplin/src/index.ts(4,27)
          TS2339: Property 'require' does not exist on type 'JoplinPlugins'.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! joplin-plugin-tohexo@1.0.0 dist: `webpack --joplin-plugin-config buildMain && webpack --joplin-plugin-config buildExtraScripts && webpack --joplin-plugin-config createArchive`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the joplin-plugin-tohexo@1.0.0 dist script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/chenwj/.npm/_logs/2021-02-22T01_39_23_366Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! joplin-plugin-tohexo@1.0.0 prepare: `npm run dist`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the joplin-plugin-tohexo@1.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/chenwj/.npm/_logs/2021-02-22T01_39_23_465Z-debug.log

```

**Comment by CalebJohn** [comment]
Do you mind sharing your code?

**Comment by ChenWenjiang** [comment]
@CalebJohn sorry for answering you until now
```
import joplin from 'api';
import { MenuItemLocation, ToolbarButtonLocation } from 'api/types';

const fs = joplin.plugins.require('fs-extra');
var stat=fs.stat;

const srcdir:string = '/home/chenwj/Documents/JoplinData';
const dstdir:string = '/home/chenwj/Projects/peacedog/source/_posts';
var copy=function(src,dst){
    fs.readdir(src,function(err,paths){
        console.log(paths)
        if(err){
            throw err;
        }
        paths.forEach(function(path){
            var _src=src+'/'+path;
            var _dst=dst+'/'+path;
            var readable;
            var writable;
            stat(_src,function(err,st){
                if(err){
                    throw err;
                }

                if(st.isFile()){
                    readable=fs.createReadStream(_src);//创建读取流
                    writable=fs.createWriteStream(_dst);//创建写入流
                    readable.pipe(writable);
                }else if(st.isDirectory()){
                    exists(_src,_dst,copy);
                }
            });
        });
    });
}

var exists=function(src,dst,callback){
		fs.exists(src,function(exist){
			if(exist){
				fs.exists(dst,function(exist){
					if(exist){
            callback(src,dst);
					}else{
            fs.mkdir(dst,function(){
                callback(src,dst)
            });
					}
				});
			}
		});
}

joplin.plugins.register({
	onStart: async function() {
		await joplin.commands.register({
			name: 'tohexo',
			label: 'Export to Hexo',
			execute: async (noteIds:string[]) => {
				const notebooks = [];
				for (const noteId of noteIds) {
					//	console.info("noteIds:",noteIds);
					const note = await joplin.data.get(['notes', noteId],{ fields: ['id','title','body','parent_id','created_time'] });
					if(!note){
						console.info('ERROR:找不到此id对应的笔记:',noteId);
						continue;
					}
					var date = new Date(note.created_time);
					var Y = date.getFullYear() + '-';
					var M = (date.getMonth()+1 < 10 ? '0'+(date.getMonth()+1) : date.getMonth()+1) + '-';
					var D = date.getDate() + ' ';
					var h = date.getHours() + ':';
					var m = date.getMinutes() + ':';
					var s = date.getSeconds(); 
					var dateString = Y+M+D+h+m+s; 

					var pid = note.parent_id;
					var l = 0;
					while(pid && pid!=''){
						l = notebooks.push(await joplin.data.get(['folders',pid],{fields:['title','parent_id']}));
						pid = notebooks[l-1].parent_id;
						//	console.info("pid:",pid);
					}
					//	console.info('notebook:',notebooks);
					const tags = await joplin.data.get(['notes', noteId,'tags']);
					//	console.info('tags:',tags);
					
					var newbody:string = 
						'---\ntitle: '+ note.title + '\n'
						+ 'date: ' +dateString + '\n' 
						+ 'tags: ' + '\n';
					for(const tag of tags.items){
						newbody += '- ' + tag.title + '\n';
					}
					newbody += 'categories:\n';
					while(l!=0){
						newbody += '- ' + notebooks.pop().title + '\n'; 
						l--;
					}
					newbody += '---\n';
					const lines = picReplace(note.body);
					for(const line of lines){
						newbody += line + '\n';
					}
					//					console.info(newbody);
					fs.writeFileSync(dstdir+'/'+note.id+'.md',newbody)
					exists(srcdir + '/edit-'+note.id, dstdir + '/'+note.id,copy)
				}
			}
		});
		await joplin.views.menuItems.create('contextMenuItem1', 'tohexo', MenuItemLocation.NoteListContextMenu);
	},
});

function picReplace(noteBody:string) {
	var isCodeSnippet = false;
	const piclines= [];
	const lines = noteBody.split('\n');
	const linesOut = [];
	for (const line of lines) {
		if(isCodeSnippet){
			if(line.slice(0,3)=='```'){
				isCodeSnippet = false;
			}
			linesOut.push(line);
		}else{
			if(line.slice(0,3)=='```'){
				isCodeSnippet = true;
			}
			const match = line.match(/!\[.*\]\(.*\)/);
			const match2 = line.match(/`.*!\[.*\]\(.*\).*`/);
			if (match && !match2){ 
				linesOut.push(line.replace(/\!\[(.*)\]\((.*\/)+(.*)\)/,'$`![$1]($3)$\''));
			}else{
				linesOut.push(line);
			}
		}	
	}
	return linesOut;
}

```

**Comment by CalebJohn** [comment]
@ChenWenjiang 

I tried your code on my computer and you're correct, the generator seems out of date and running `npm update` does not seem to help. For now I think you'll need to manually copy the api files from [here](https://github.com/laurent22/joplin/tree/dev/packages/generator-joplin/generators/app/templates/api)

@laurent22 I checked the [generator page on npm](https://www.npmjs.com/package/generator-joplin) and it seems the last update was on January 15. I think I know how to publish an update, I can give it a try if you don't mind. That is unless you have a reason to delay the release.
